src/imagenes/extract.py

```diff
@@ -212,103 +197,124 @@ def parse(self, dir3, fname):
-            print("img", img)
+    @staticmethod
+    def parse_html(html, choosen_pages):
+        soup = bs4.BeautifulSoup(html, features="html.parser")
+
+        new_images = set()
+
+        for img_tag in soup.find_all('img'):
 
-        # reemplazamos ancho y alto por un fragment en la URL de la imagen
-        msize = WIDTH_HEIGHT.search(p3)
-        p3 = WIDTH_HEIGHT.sub("", p3)
-        web_url = 'http:' + img
+            dsk_url, web_url = ImageParser.replace(img_tag)
+            if dsk_url:
+                new_images.add((dsk_url, web_url))
 
-        if img.startswith("//upload.wikimedia.org/wikipedia/commons/"):
-            partes = img[33:].split("/")
+        for a_tag in soup.find_all('a'):
+            ImageParser.fixlinks(a_tag, choosen_pages)
+
+        html = unicode(soup)
+        return html, new_images
+
+    @staticmethod
+    def replace(tag):
+        """Replaces the img tag in place and return the disk and web urls"""
+        img_src = tag.attrs.get("src")
+
+        web_url = 'http:' + img_src
+
+        if img_src.startswith("//upload.wikimedia.org/wikipedia/commons/"):
+            partes = img_src[33:].split("/")
             if len(partes) == 6:
                 del partes[4]
             elif len(partes) == 4:
                 pass
             else:
-                raise ValueError("Strange image format! %r" % img)
+                raise ValueError("Strange image format! %r" % img_src)
 
             dsk_url = "/".join(partes)
 
-        elif img.startswith("//bits.wikimedia.org/"):
-            dsk_url = img[46:]
+        elif img_src.startswith("//bits.wikimedia.org/"):
+            dsk_url = img_src[46:]
 
-        elif img.startswith("//upload.wikimedia.org/wikipedia/es/"):
-            dsk_url = img[36:]
+        elif img_src.startswith("//upload.wikimedia.org/wikipedia/es/"):
+            dsk_url = img_src[36:]
 
-        elif img.startswith("//upload.wikimedia.org/"):
-            dsk_url = img[23:]
+        elif img_src.startswith("//upload.wikimedia.org/"):
+            dsk_url = img_src[23:]
 
-        elif img.startswith("/w/extensions/"):
-            web_url = WIKIPEDIA_URL + img
-            dsk_url = img[3:]
+        elif img_src.startswith("/w/extensions/"):
+            web_url = WIKIPEDIA_URL + img_src
+            dsk_url = img_src[3:]
 
-        elif img.startswith("https://wikimedia.org/api/rest_v1/media/"):
-            web_url = img
-            dsk_url = img[40:]
-
-        elif img.startswith("/api/rest_v1/page/"):
-            web_url = WIKIPEDIA_URL + img
-            dsk_url = img[18:]
+        elif img_src.startswith("https://wikimedia.org/api/rest_v1/media/"):
+            web_url = img_src
+            dsk_url = img_src[40:]
 
+        elif img_src.startswith("/api/rest_v1/page/"):
+            web_url = WIKIPEDIA_URL + img_src
+            dsk_url = img_src[18:]
+        elif any((to_remove in img_src for to_remove in IMAGES_TO_REMOVE)):
+            tag.extract()
```

Comment:
Por qué nos interesa modificar el tag acá si devolvemos None, None (con lo cual nada sería utilizado... creo)?

Reply:
Los None, None son para que no se descargue la imágen en un paso siguiente, pero si no sacamos el tag entonces el html final va a contener este `<img>`. Los tags que reciben estas funciones se modifican "in place", y al final de todo con "el soup" se genera el html nuevo que despues se usa en pasos siguientes.